projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,330 @@
+# 1. Introduction
+
+This document outlines the remote procedure call (RPC) methods for a decentralized application (dApp) to interact with Hathor wallets.
+
+The design is currently a minimum viable product (MVP), addressing only known needs from our community. It should be noted that this API design may change as the platform evolves and additional needs arise.
+
+The idea here is to partially replicate the headless wallet API as it is already being used by many partners and has been validated to work on their use cases
+
+# 2. Reference-level explanation
+
+## 2.1 RPC Methods
+
+### 2.1.1 `htr_createTx`
+
+The `htr_createTx` method creates a new transaction based on parameters. It will select the `utxos` automatically, mine and send the transaction
+
+**Parameters**
+
+1. `outputs` - An array of outputs
+	1. `address` - Destination address of the output. Required if P2PKH or P2SH
+	2. `value` - (Required if P2PKH or P2SH) The value parameter must be an integer with the value in cents, i.e., 123 means 1.23 HTR.
+	3. `token` - (Defaults to `HTR`) Token id of the output.
+	4. `type` - (Required if data script output) Type of output script.
+	5. `data` - (Required if data script output) Data string of the data script output.
+2. `inputs` - (Optional) Inputs to create the transaction
```

Comment:
Will we support the filters like in the headless? I mean, so we can specify which addresses to spend from.

Reply:
We should, updated the design in e64e5fb7d52b96e35bca5779e852d8d21b7a1227